### ISSUE TITLE
Bedrock Player dies from fall damage when walking down block staircase

### ISSUE THREAD

**Issue by Kasi07** [GeyserMC/Geyser#5127]
### Describe the bug

A player walking down a block staircase, he sometimes gets a little fall damage but even if not finally dies from fall damage.
This even happens if carefully and slowly walking down the stairs.

### To Reproduce

1. Build a block staircase
2. Walk down

### Expected behaviour

The player does not get any fall damage.

### Screenshots / Videos

https://github.com/user-attachments/assets/9b026ced-f72a-4902-88b2-a8d226ca861d

### Geyser Dump

https://dump.geysermc.org/5rQNsHFvOvpMSdt6622WWKIVufHmhaWH

### Geyser Version

2.5.0-b706 (git-master-6983095)

### Minecraft: Bedrock Edition Device/Version

1.21.44, Windows

**Comment by Kasi07** [comment]
As mentioned in the issue #5129 these two issues relate to each other.

I tested the issue regarding onGround mentioned in #5129 in relation to this issue. Players walking down a slab staircase, that is higher than 3 Blocks, take fall damage when they walk off the slab onto a full block.

Regarding the example video I provided initially, I think it is very plausible that this also relates to the movement auth change.
